src/sigmatcher/definitions.py

```diff
@@ -49,6 +67,39 @@ def check_strings(self, strings: List[str]) -> List[str]:
     def capture(self, value: str) -> List[str]:
         return self.signature.findall(value)
 
+    def get_dependencies(self) -> List[str]:
+        return [macro.rpartition(".")[0] for macro in self._get_raw_macros]
+
+    @cached_property
+    def _get_raw_macros(self) -> Set[str]:
+        return set(self.MACRO_REGEX.findall(self.signature.pattern))
+
+    def resolve_macros(self, results: Dict[str, Union[Result, Exception, None]]) -> Self:
+        if not self._get_raw_macros:
+            return self
+
+        new_pattern = self.signature.pattern
+        for macro in self._get_raw_macros:
+            result_identifier, _, result_modifier = macro.rpartition(".")
+            result = results[result_identifier]
+            assert result is not None
+            assert not isinstance(result, Exception)
+
+            if result_modifier == "name":
+                resolved_macro = result.new.name
+            elif result_modifier == "java":
+                resolved_macro = result.new.to_java_representation()
+            elif result_modifier == "package":
+                if not isinstance(result, MatchedClass):
+                    raise ValueError("Only classes can be used with the 'package' modifier.")
+                resolved_macro = result.new.pacakge
+            else:
+                raise ValueError(f"Unknown macro modifier: {result_modifier}")
```

Comment:
```suggestion
            try:
                resolved_macro = getattr(result.new, result_modifier)
            except AttributeError:
                raise ValueError(f"Invalid macro modifier: {result_modifier} for {result.new.__class__.__name__}")
```

And make the result modifier `java` a `@property` of `Result`?

Reply:
Sounds like a good idea